infrastructure/src/main/java/org/corfudb/infrastructure/CorfuServerNode.java

```diff
@@ -67,32 +66,37 @@
      * @param serverContext Initialized Server Context.
      */
     public CorfuServerNode(@Nonnull ServerContext serverContext) {
-        this(serverContext,
-                ImmutableMap.<Class, AbstractServer>builder()
-                        .put(BaseServer.class, new BaseServer(serverContext))
-                        .put(SequencerServer.class, new SequencerServer(serverContext))
-                        .put(LayoutServer.class, new LayoutServer(serverContext))
-                        .put(LogUnitServer.class, new LogUnitServer(serverContext))
-                        .put(ManagementServer.class, new ManagementServer(serverContext))
-                        .build()
-        );
+
+        this(serverContext, createServerMap(serverContext));
     }
 
-    /**
-     * Corfu Server initialization.
-     *
-     * @param serverContext Initialized Server Context.
-     * @param serverMap     Server Map with all components.
-     */
-    public CorfuServerNode(@Nonnull ServerContext serverContext,
-                           @Nonnull Map<Class, AbstractServer> serverMap) {
+    public CorfuServerNode(@NonNull ServerContext serverContext,
+                           @NonNull Map<Class, AbstractServer> serverMap) {
         this.serverContext = serverContext;
         this.serverMap = serverMap;
-        router = new NettyServerRouter(new ArrayList<>(serverMap.values()));
+        this.router = new NettyServerRouter(new ArrayList<>(serverMap.values()));
         this.serverContext.setServerRouter(router);
         this.close = new AtomicBoolean(false);
     }
 
+    /**
+     * Given a server context create a map from class to a server.
+     *
+     * @param context Corfu server context.
+     * @return A map.
+     */
+    private static ImmutableMap<Class, AbstractServer> createServerMap(ServerContext context) {
+        ServerConfigurator configurator = new ServerConfigurator(context);
+
+        return ImmutableMap.<Class, AbstractServer>builder()
+                .put(BaseServer.class, new BaseServer(context))
+                .put(SequencerServer.class, new SequencerServer(context))
+                .put(LayoutServer.class, new LayoutServer(context))
```

Comment:
Can you move the BaseServer, SequencerServer and LayoutServer to the configurator ?

Since theres no transformation needed and only the context is being passed to the constructor I don't see how this will be an extra 1000 lines?